ValveKeyValue/ValveKeyValue/Deserialization/KVAppendingObjectBuilder.cs

```diff
@@ -15,6 +15,12 @@ protected override void FinalizeState()
         {
             base.FinalizeState();
 
+            if (StateStack.Count <= 0)
```

Comment:
For the sake of simplicity, as you said some changes from other PR applied here, could you please leave this PR just to the nesting fix.

Reply:
Done, though this logic will ultimately have to be added back in at some point.